tests/tests.js

```diff
@@ -28,144 +28,217 @@ exports.defineAutoTests = () => {
+
+      let numOfNativeCalls = cordova.exec.calls.count() - window.console.error.calls.count();
+      expect(numOfNativeCalls).toBe(4);
+      expect(window.NewRelic.setAttribute).toHaveBeenCalledTimes(11);
     });
 
     it('should remove attribute', () => {
-      spyOn(window.NewRelic, "removeAttribute");
+      spyOn(window.NewRelic, "removeAttribute").and.callThrough();
       window.NewRelic.removeAttribute('eventType');
       expect(window.NewRelic.removeAttribute).toHaveBeenCalledTimes(1);
     });
 
     it('should record breadcrumb', () => {
-      spyOn(window.NewRelic, "recordBreadcrumb");
+      spyOn(window.NewRelic, "recordBreadcrumb").and.callThrough();
+      spyOn(cordova, "exec").and.callThrough();
+      spyOn(window.console, "error").and.callThrough();
+
       window.NewRelic.recordBreadcrumb('testName', { test: 123, valid: 'yes' });
-      expect(window.NewRelic.recordBreadcrumb).toHaveBeenCalledTimes(1);
-    });
 
-    it('should record custom event', () => {
-      spyOn(window.NewRelic, "recordCustomEvent");
-      window.NewRelic.recordCustomEvent('eventType', 'eventName', { test: 123, valid: 'yes' });
-      window.NewRelic.recordCustomEvent('eventType', '', { test: 123, valid: 'yes' });
-      window.NewRelic.recordCustomEvent('eventType', undefined, { test: 123, valid: 'yes' });
-      expect(window.NewRelic.recordCustomEvent).toHaveBeenCalledTimes(3);
+      // Bad argument
+      window.NewRelic.recordBreadcrumb(null, { test: 123, valid: 'no' });
+
+      let numOfNativeCalls = cordova.exec.calls.count() - window.console.error.calls.count();
+      expect(numOfNativeCalls).toBe(1);
+      expect(window.NewRelic.recordBreadcrumb).toHaveBeenCalledTimes(2);
     });
 
+    // it('should record custom event', () => {
+    //   spyOn(window.NewRelic, "recordCustomEvent").and.callThrough();
+    //   spyOn(cordova, "exec").and.callThrough();
+    //   spyOn(window.console, "error").and.callThrough();
+
+    //   window.NewRelic.recordCustomEvent('eventType', 'eventName', { test: 123, valid: 'yes' });
+    //   window.NewRelic.recordCustomEvent('eventType', '', { test: 123, valid: 'yes' });
+    //   window.NewRelic.recordCustomEvent('eventType', undefined, { test: 123, valid: 'yes' });
+
+    //   // Bad arguments
+    //   window.NewRelic.recordCustomEvent('eventType', null, { test: 123, valid: 'yes' });
+    //   window.NewRelic.recordCustomEvent('eventType', [], { test: 123, valid: 'yes' });
+    //   window.NewRelic.recordCustomEvent('eventType', {}, { test: 123, valid: 'yes' });
+
+    //   // Every console call creates a custom event
+    //   let numConsoleErrCalls = window.console.error.calls.count();
+    //   let numOfNativeCalls = cordova.exec.calls.count() - numConsoleErrCalls;
+    //   let totalCustomEventCalls = window.NewRelic.recordCustomEvent.calls.count() - numConsoleErrCalls;
+    //   expect(numOfNativeCalls).toBe(3);
+    //   expect(totalCustomEventCalls).toBe(6);
+    // });
```

Comment:
This also works and reports a custom event in NR1. But the test harness app from cordova-paramedic didn't like this for some reason. Stack trace below:
```
Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x1ae50fe60 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x1ae5623c0 pthread_kill + 256
2   libsystem_c.dylib             	       0x180129e38 abort + 124
3   libc++abi.dylib               	       0x18023b0c0 abort_message + 128
4   libc++abi.dylib               	       0x18022b964 demangling_terminate_handler() + 300
5   libobjc.A.dylib               	       0x180032cc0 _objc_terminate() + 124
6   libc++abi.dylib               	       0x18023a570 std::__terminate(void (*)()) + 16
7   libc++abi.dylib               	       0x18023cd48 __cxxabiv1::failed_throw(__cxxabiv1::__cxa_exception*) + 32
8   libc++abi.dylib               	       0x18023cd08 __cxa_throw + 132
9   libobjc.A.dylib               	       0x18005144c objc_exception_throw + 360
10  CoreFoundation                	       0x18041afbc -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
11  CoreFoundation                	       0x180410204 ___forwarding___ + 1308
12  CoreFoundation                	       0x18041268c _CF_forwarding_prep_0 + 92
13  NewRelic                      	       0x100778f18 +[NewRelic recordCustomEvent:name:attributes:] + 124
14  HelloCordova                  	       0x1000bdf7c -[NewRelicCordovaPlugin recordCustomEvent:] + 284 (NewRelicCordovaPlugin.m:45)
15  HelloCordova                  	       0x1000c21a8 -[CDVCommandQueue execute:] + 720 (CDVCommandQueue.m:181)
16  HelloCordova                  	       0x1000c90b8 -[CDVWebViewEngine userContentController:didReceiveScriptMessage:] + 272 (CDVWebViewEngine.m:462)
17  HelloCordova                  	       0x1000ca2e8 -[CDVWebViewWeakScriptMessageHandler userContentController:didReceiveScriptMessage:] + 116 (CDVWebViewEngine.m:604)
18  WebKit                        	       0x103610344 ScriptMessageHandlerDelegate::didPostMessage(WebKit::WebPageProxy&, WebKit::FrameInfoData&&, API::ContentWorld&, WebCore::SerializedScriptValue&) + 188
19  WebKit                        	       0x1039542e8 WebKit::WebUserContentControllerProxy::didPostMessage(WTF::ObjectIdentifier<WebKit::WebPageProxyIdentifierType>, WebKit::FrameInfoData&&, unsigned long long, WTF::Span<unsigned char const, 18446744073709551615ul> const&, WTF::CompletionHandler<void (WTF::Span<unsigned char const, 18446744073709551615ul> const&, WTF::String const&)>&&) + 564
20  WebKit                        	       0x103c20448 void IPC::handleMessageAsync<Messages::WebUserContentControllerProxy::DidPostMessage, WebKit::WebUserContentControllerProxy, void (WebKit::WebUserContentControllerProxy::*)(WTF::ObjectIdentifier<WebKit::WebPageProxyIdentifierType>, WebKit::FrameInfoData&&, unsigned long long, WTF::Span<unsigned char const, 18446744073709551615ul> const&, WTF::CompletionHandler<void (WTF::Span<unsigned char const, 18446744073709551615ul> const&, WTF::String const&)>&&)>(IPC::Connection&, IPC::Decoder&, WebKit::WebUserContentControllerProxy*, void (WebKit::WebUserContentControllerProxy::*)(WTF::ObjectIdentifier<WebKit::WebPageProxyIdentifierType>, WebKit::FrameInfoData&&, unsigned long long, WTF::Span<unsigned char const, 18446744073709551615ul> const&, WTF::CompletionHandler<void (WTF::Span<unsigned char const, 18446744073709551615ul> const&, WTF::String const&)>&&)) + 2180
21  WebKit                        	       0x103c1fbb0 WebKit::WebUserContentControllerProxy::didReceiveMessage(IPC::Connection&, IPC::Decoder&) + 76
22  WebKit                        	       0x1036e599c IPC::MessageReceiverMap::dispatchMessage(IPC::Connection&, IPC::Decoder&) + 212
23  WebKit                        	       0x1038c12b4 WebKit::WebProcessProxy::didReceiveMessage(IPC::Connection&, IPC::Decoder&) + 36
24  WebKit                        	       0x1036df56c IPC::Connection::dispatchMessage(std::__1::unique_ptr<IPC::Decoder, std::__1::default_delete<IPC::Decoder> >) + 640
25  WebKit                        	       0x1036dedac IPC::Connection::dispatchIncomingMessages() + 616
26  JavaScriptCore                	       0x101780ba4 WTF::RunLoop::performWork() + 172
27  JavaScriptCore                	       0x1017815d8 WTF::RunLoop::performWork(void*) + 32
28  CoreFoundation                	       0x180371070 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
29  CoreFoundation                	       0x180370fb8 __CFRunLoopDoSource0 + 172
30  CoreFoundation                	       0x180370728 __CFRunLoopDoSources0 + 232
31  CoreFoundation                	       0x18036ae68 __CFRunLoopRun + 756
32  CoreFoundation                	       0x18036a75c CFRunLoopRunSpecific + 584
33  GraphicsServices              	       0x188f60c98 GSEventRunModal + 160
34  UIKitCore                     	       0x10b5d0b74 -[UIApplication _run] + 868
35  UIKitCore                     	       0x10b5d4b1c UIApplicationMain + 124
36  HelloCordova                  	       0x1000bd564 main + 56 (main.m:32)
37  dyld_sim                      	       0x100455fa0 start_sim + 20
38  dyld                          	       0x10017d08c start + 520
```